src/ui-kit/experimental/listbox/listbox.component.ts

```diff
@@ -0,0 +1,288 @@
+import {
+  Component, Output,
+  forwardRef, Input,
+  EventEmitter, ElementRef, ViewChild
+} from '@angular/core';
+import {
+  FormControl,
+  ControlValueAccessor,
+  NG_VALUE_ACCESSOR
+} from '@angular/forms';
+import { KeyHelper, KEYS } from '../../utilities/key-helper/key-helper';
+import { FieldsetWrapper } from '../../wrappers/fieldset-wrapper';
+
+export interface OptionModel {
+  name: string;
+  value: string;
+  label: any;
+  required: boolean;
+  checked: boolean;
+  disabled: boolean;
+}
+@Component({
+  selector: 'sam-listbox',
+  templateUrl: './listbox.component.html',
+  styleUrls: ['./listbox.component.scss'],
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => SamListBoxComponent),
+    multi: true
+  }]
+})
+
+
+export class SamListBoxComponent implements ControlValueAccessor {
+  /**
+  * Deprecated, Sets the bound value of the component
+  */
+  @Input() model: any = [];
+  /**
+  * Sets the array of checkbox values and labels (see OptionsType[])
+  */
+  @Input() options: OptionModel[];
+  /**
+  * Sets the label text
+  */
+  @Input() label: string;
+  /**
+  * Sets the semantic description for the component
+  */
+  @Input() name: string;
+  /**
+  * Sets helpful text for the using the component
+  */
+  @Input() hint: string;
+  /**
+  * Sets required text on component
+  */
+  @Input() required: boolean = false;
+  /**
+  * Sets the form control error message
+  */
+  @Input() errorMessage: string;
+  /**
+  * Sets the angular FormControl
+  */
+  @Input() control: FormControl;
+  /**
+  * Sets disabled state
+  */
+  @Input() disabled: boolean;
```

Comment:
@yerramshilpa Can you remove this and let it be set by the `setDisabledState` method? Also, why can't you bind directly to disabled in the HTML template?

Reply:
@colinalford , I am already doing in the HTML directly. I will remove it from here.